services/solr-names-updater/src/solr_names_updater/names_processors/possible_conflicts.py

```diff
@@ -0,0 +1,103 @@
+                                                 find_name_by_name_states,
+                                                 post_to_solr_feeder)
+
+
+def process_add_to_solr(state_change_msg: dict):  # pylint: disable=too-many-locals, , too-many-branches
+    """Process possible conflicts update via Solr feeder api."""
+    logger.debug('names processing: %s', state_change_msg)
+    nr_num = state_change_msg.get('nrNum')
+    nr = RequestDAO.find_by_nr(nr_num)
+    send_to_solr_add(nr)
+
+
+def process_delete_from_solr(state_change_msg: dict):  # pylint: disable=too-many-locals, , too-many-branches
+    """Process possible conflicts update via Solr feeder api."""
+    logger.debug('names processing: %s', state_change_msg)
+    nr_num = state_change_msg.get('nrNum')
+    nr = RequestDAO.find_by_nr(nr_num)
+    send_to_solr_delete(nr)
+
+
+def send_to_solr_add(nr: RequestDAO):
+    """Send json payload to add possible conflict from solr for NR."""
+    name_states = [NameState.APPROVED.value, NameState.CONDITION.value]  # pylint: disable=no-member
+    names = find_name_by_name_states(nr.id, name_states)
+    name = names[0]
+    jur = nr.xproJurisdiction if nr.xproJurisdiction else 'BC'
+    payload_dict = construct_payload_dict(nr, name, jur)
+
+    resp = post_to_solr_feeder(payload_dict)
+    if resp.status_code != 200:
+        logger.error("""failed to add possible conflict to solr for %s,
+                    status code: %i, error reason: %s, error details: %s""",
+                     nr.nrNum,
+                     resp.status_code,
+                     resp.reason,
+                     resp.text)
+
+
+def send_to_solr_delete(nr: RequestDAO):
+    """Send json payload to delete possible conflict from solr for NR."""
+    payload_dict = {
+        'solr_core': 'possible.conflicts',
+        'request': {
+            'delete': [nr.nrNum],
+            'commit': {}
+        }
+    }
+
+    request_str = json.dumps(payload_dict['request'], indent=4)
+    payload_dict['request'] = request_str
+    resp = post_to_solr_feeder(payload_dict)
+    if resp.status_code != 200:
+        logger.error("""failed to delete possible conflict from solr for %s,
+                     status code: %i, error reason: %s, error details: %s""",
+                     nr.nrNum,
+                     resp.status_code,
+                     resp.reason,
+                     resp.text)
+
+
+def construct_payload_dict(nr: RequestDAO, name, jur):
+    """Construct json payload used to invoke solr feeder endpoint for adding possible conflicts for a given NR."""
+    payload_dict = {'solr_core': 'possible.conflicts'}
+    payload_request = {}
+    start_date = convert_to_solr_conformant_datetime_str(nr.submittedDate)
+    payload_request['add'] = {
+        'doc': {
+            'id': nr.nrNum,
+            'name': name.name,
+            'state_type_cd': name.state,
+            'source': '',  # todo determine what value should be here
```

Comment:
what value should the `source` field be?

Reply:
probably 'solr-names-updater' -- unless this is for NRO vs NAMEX (I don't think it matters too much since this isn't used anywhere that I know of). Thor might remember something I don't though

Reply:
Actually I think it should probably be the request._source even though for now it is usually incorrect

Reply:
updated to use `nr.source`